### ISSUE TITLE
why does this C code  always give me null detection? 

### ISSUE THREAD

**Issue by goodtogood** [AlexeyAB/darknet#589]
Hi @AlexeyAB 
the code below is almost copied from test_detector function of detector.c,
I just made a little modifaction, It always gives me nothing.
Is there something I missed?
thanks a lot !

```
#include "network.h"
#include "unistd.h"
#include "parser.h"
#include "region_layer.h"


int main(int argc, char* argv[])
{
	char* namefile = "./data/voc.names";
	char* cfgfile = "./tiny-yolo-voc.cfg";
	char* weightfile = "./tiny-yolo-voc.weights";
	int gpu_index = 1;
	char* input = "./data/dog.jpg";
	float thresh = .24;
	float nms = .3;

	char** names = get_labels(namefile);

	image** alphabet = load_alphabet();

	cuda_set_device(gpu_index);

	network net = parse_network_cfg_custom(cfgfile, 1);
	if (weightfile)
	{
		load_weights(&net, weightfile);
	}
	set_batch_network(&net, 1);

	srand(2222222);

	image im = load_image_color(input, 0, 0);

	image sized = resize_image(im, net.w, net.h);

	layer l = net.layers[net.n - 1];

	box* boxes = calloc(l.w * l.h * l.n, sizeof(box));
	float** probs = calloc(l.w * l.h * l.n, sizeof(float *));
	for (int j = 0; j < l.w * l.h * l.n; ++j) probs[j] = calloc(l.classes, sizeof(float *));

	float* X = sized.data;
	network_predict(net, X);

	save_image(sized, "resized");

	get_region_boxes(l, 1, 1, thresh, probs, boxes, 0, 0);
	if (nms) do_nms_sort(boxes, probs, l.w * l.h * l.n, l.classes, nms);


	draw_detections(im, l.w * l.h * l.n, thresh, boxes, probs, names, alphabet, l.classes);
	save_image(im, "predicted");

	free_image(im);
	free_image(sized);
	free(boxes);
	return 0;
}
```

just print network information and quit.

![0](https://user-images.githubusercontent.com/10088733/38433424-640afc64-39fd-11e8-8950-8498e67466d5.jpg)

**Comment by AlexeyAB** [comment]
Try to add `#include "image.h"`

How old your repo? 
In the last version I use code that supports Yolo v2 and Yolo v3: https://github.com/AlexeyAB/darknet/blob/2a9f7e44ce1b73d3d56ef83f83e94f074ecac3f9/src/detector.c#L1062-L1065
Instead of 
```cpp
get_region_boxes(l, 1, 1, thresh, probs, boxes, 0, 0);
	if (nms) do_nms_sort(boxes, probs, l.w * l.h * l.n, l.classes, nms);
	draw_detections(im, l.w * l.h * l.n, thresh, boxes, probs, names, alphabet, l.classes);
```

**Comment by goodtogood** [comment]
Hi, @AlexeyAB 
yes, I did use the old repo before v3,  
cuz I found results detected of v3 are different from v2,
the number of boxes detected of v3 is always fewer than v2,
but image ,model and cfg are the same one. I don't know why, 
as for fps, It seems that the performance of v3 is a little slower than v2.

![v2](https://user-images.githubusercontent.com/10088733/38451834-be4d4b34-3a69-11e8-8aab-c97e3f3af9f7.jpg)



![v3](https://user-images.githubusercontent.com/10088733/38451833-b75d80c8-3a69-11e8-9370-b0f06a8324f8.jpg)

**Comment by AlexeyAB** [comment]
@goodtogood 

> yes, I did use the old repo before v3,
> cuz I found results detected of v3 are different from v2,
> the number of boxes detected of v3 is always fewer than v2,
> but video ,model and cfg are the same one. I don't know why,
> as for fps, It seems that the performance of v3 is a little slower than v2.

Do you use the same Yolo v2 model on both Repo-v2 and Repo-v3?

About number of boxes, there was changes NMS from 0.4 to 0.45: https://github.com/AlexeyAB/darknet/commit/d9ae3dd681ed1c98e807ff937dbbb9cfc4d19fe0#diff-4e71a2cf0098713e52e5dae1dfd56c06R81

On my PC Repo-v3 a little bit faster than Repo-v2 due to fused layers (Batch-norm + Conv)

**Comment by goodtogood** [comment]
Hi, @AlexeyAB 
yes , I used the same v2 model on both repo v2 and v3 ,
the cfg  is  based on yolo-voc.2.0.cfg, 
I compared yolo-voc.2.0.cfg with yolov2-voc.cfg,
I found there's some difference as below,

![yolo-voc2 0-vs-yolov2-voc](https://user-images.githubusercontent.com/10088733/38454160-1f7e0b08-3a94-11e8-9387-a141ffb125c7.jpg)

does it have an effect on old v2 models ?
I  noticed NMS was changed, so I've changed it to 0.4.
but It seems the number of boxes keeps the same as without it.

I'll clone a brand new repo-v3 and try again.

thank you !
